src/goto-symex/symex_atomic_section.cpp

```diff
@@ -16,10 +16,9 @@ void goto_symext::symex_atomic_begin(statet &state)
   if(state.guard.is_false())
     return;
 
-  // we don't allow any nesting of atomic sections
-  if(state.atomic_section_id!=0)
-    throw "nested atomic section detected at "+
-      state.source.pc->source_location.as_string();
+  INVARIANT(
+  	state.atomic_section_id == 0, 
+  	"we don't support nesting of atomic sections");
```

Comment:
Indentation is incorrect. However, is this actually an INVARIANT, or is this a user error? e.g. is this something that could be triggered by a suitably written input program? I note that the old throw code includes user source code location info, which suggests it's (potentially) a user level error. Either way, this part of the error string is not included in the INVARIANT, which potentially makes debugging harder.